### ISSUE TITLE
[Feature Request] Background Worker for jobs (Push Notifications)

### ISSUE THREAD

**Issue by vzool** [directus/directus#2226]
Hi,

I'm looking for away to do some long jobs like (Custom mail sending,  SMS calls, Push Notifications, report generation and any remote service calls). It there a way to make this and I'm not aware of?

Thanks

**Comment by dekyfin** [comment]
Is there any reason you cannot use cron for those tasks?

**Comment by shealavington** [comment]
This sounds like something you would create specifically for your application, as mentioned above, CRON could be used for scheduled reporting tasks and Hooks could be used for notifications.

However, on the other hand, this does sound like a great idea for a "**Task Scheduler Extension**". The extension could provide an interface to build a "task" and schedule that to run, which I can see being useful for some situations where you rely on the data. 

### Examples: 
- Every 30 days, loop each user who has email notifications enabled, select all "payment receipts" for them, display the data in a "twig template" and send it to each users email address.
- Every day at midnight, clear all activity more than 7 days ago (this keeps exactly 1 week of logs), and provides a friendly interface for keeping the database clean.
- When a user updates their personal information, send a push-notification
- Etc.

🤔

**Comment by vzool** [comment]
Yes, I think I can do that @dekyfin. Using cron, which is simple and very effective.
But, I don't know what is the entry point for this. 🤔

**Comment by vzool** [comment]
@rijkvanzanten The use of Hooks is easy but expensive for a long run. I really do prefer to make API calls simple and fast as much as I can. So I consider to dedicate API endpoints to handle data only while any time consuming/dependable operations left for the "**Background Tasks**". 🙂

**Comment by vzool** [comment]
@shealavington Yes, I need "**Task Scheduler**". This is what I'm looking for. 😍 
The scheduler always waiting and looking for any scheduled/new task to run, this will help many many other to gain advantages of "**Parallelism-Like**" on directus. 🤓

**Comment by vzool** [comment]
> We had a PR for a "Cron Job Manager" before, but I got so busy with other stuff didn't have time to review and then we release version 7.
> 
> As @shealavington said this can be done with cron job.
> 
> This is a nice feature to have. but as right now it can be done by using cron job to run an specific script, or instantiate a worker queue service in hooks to execute your custom hook.

Which one? 😃
